sample-compose/app/src/main/java/co/nimblehq/sample/compose/ui/screens/home/HomeScreen.kt

```diff
@@ -19,26 +19,34 @@ import co.nimblehq.sample.compose.ui.screens.AppBar
 import co.nimblehq.sample.compose.ui.theme.ComposeTheme
 import co.nimblehq.sample.compose.ui.userReadableMessage
 import com.google.accompanist.permissions.*
+import kotlinx.coroutines.delay
 
 @Composable
 fun HomeScreen(
     viewModel: HomeViewModel = hiltViewModel(),
     navigator: (destination: AppDestination) -> Unit
 ) {
+    val showLoading: IsLoading by viewModel.showLoading.collectAsState()
+    val uiModels: List<UiModel> by viewModel.uiModels.collectAsState()
+    val firstTimeLaunch: Boolean by viewModel.firstTimeLaunch.collectAsState()
+
     val context = LocalContext.current
     LaunchedEffect(viewModel.error) {
```

Comment:
[Optional]
@lydiasama I know this is not your changes code 😅  but since you used `collectAsState`, I think it may be better if we can take advantage of the it as the same of others `showLoading` `uiModels` `firstTimeLaunch` in here with the `error` 👏 
```suggestion
    val error: Throwable? by viewModel.error.collectAsState(initial = null)
        LaunchedEffect(error) {
        error?.let {
            val message = it.userReadableMessage(context)
            Toast.makeText(context, message, Toast.LENGTH_SHORT).show()
        }
    }
```

Reply:
@thiennguyen0196 I don't think the `error` stream is like others as it's more like the trigger function to show the toast message, not to update or recomposition any widget here. 🤔 And also if we change it to state, it will have only `null` or `unit`, which does not have any benefit IMO.

Reply:
@lydiasama sorry for my late response as I was doing other tasks 😅 Since this is just optional and your explanation was quite clear enough, I think it is okay for me too 👍